mesa_geo/geoagent.py

```diff
@@ -32,19 +31,6 @@ def step(self):
         """Advance one step."""
         pass
 
-    def __geo_interface__(self):
-        """Return a GeoJSON Feature.
```

Comment:
By definition, the output of `__geo_interface__` seems to be JSON-serializable. If it is not, then the problem is in the implementation. Are you sure this is not needed elsewhere? You could at least just not use `__geo_interface__` in the visualization `render()`, without deleting this method. I haven't used Mesa-Geo much to be able to tell which should be the case.

Reply:
It was used in the `MapModule.render()` function, and the `GeoSpace.__geo_interface__` function, which was not used and similarly deleted. I guess it's a similar idea from [geopandas.GeoDataFrame.\_\_geo_interface\_\_](https://geopandas.org/en/stable/docs/reference/api/geopandas.GeoDataFrame.__geo_interface__.html)

There're three problems if we implement this method properly:
1. By definition all attributes of an agent should be json serialised. Each agent has a `model` attribute, i.e., `agent.model`, which is currently serialised as `str(self.model)`(but why?). Plus, the users need to make sure that all additional fields in their self-defined `GeoAgent` are json serialisable.
2. Not all attributes are necessary for frontend visualisation. This links to https://github.com/projectmesa/mesa-geo/issues/32 and is the motivation of this PR. The necessary properties for visualisation are set by `MapModule.portrayal_method()` which is outside `GeoAgent`.
3. This method is not used anywhere else apart from `MapModule.render()`.

Options:
1. We fix it, and keep it. This might leads to further issues from users (or not).
2. We do nothing and keep it. Again this might leads to further issues from users (or not).
3. We delete it, bring it back in the future if needed, and fix it then.

Do you have any suggestions?

Reply:
`self.model` is probably a complex beast, and hence why only the `str` of it becomes its serialized version.
From reading that geopandas.org documentation, I see that `__geo_interface__` is needed for GeoJSON serialization in the future, and hence, not necessarily for visualization. We might want the simulation state (which includes the agents) to be serializable, so that it can be resumed at a different time, on a different machine.

Reply:
To keep this PR small and hopefully mergeable soon, I think we should just do nothing about `__geo_interface__` for now. We can delete/fix it in a separate PR.

Reply:
Sure, I've added them back.